### DISCUSSION TITLE
mod.lua fix help

### DISCUSSION THREAD

**Discussion by glowsoony** [KadeArchive/Kade-Engine#1183] [Q&A]
welp who would have gussed it wouldnt work? please help i used this code to try to make the arrows move up and down like a wave.
see whats wrong here.

ffunction start (song)
        print("Succesfully initialized our song (Blammed)")
end

function Update (elapsed)
    local currentBeat = (songPos / 1000)*(bpm/172)
        for i=0,7 do
        setActorY(defualtStrum0Y + 10 * math.cos((currentBeat + i*0.25) * math.pi), i)
    end

end

function beatHit (beat)
        print("Da beat has hit")
end

function septHit (beat)

end

function KeyPreesed (key)

this happens too 

![image](https://user-images.githubusercontent.com/84847356/124492123-0be55d00-dd82-11eb-84b6-b21d8d21d1c8.png)

**Comment by isakube** [comment]
please repaste that with backticks ` so i can read it

**Comment by isakube** [comment]
``` lua
function start (song)
    print("Succesfully initialized our song (Blammed)")
end

function Update (elapsed)
    local currentBeat = (songPos / 1000)(bpm/172)
    for i=0,7 do
        setActorY(defaultStrum0Y + 10 * math.cos((currentBeat + i0.25) * math.pi), i)
    end
end

function beatHit (beat)
    print("Da beat has hit")
end

function septHit (beat)

end

function KeyPressed (key)

end
```

fixed lol

**Reply by glowsoony** [reply]
welp let me show you how that went XD



![image](https://user-images.githubusercontent.com/84847356/124502444-836eb880-dd91-11eb-9970-63d8a063c991.png)
 

the code might be buged lol but i am sure there is a way to fix it.

**Reply by isakube** [reply]
``` lua
setActorY(defaultStrum0Y + 10 * math.cos((currentBeat + i0.25) * math.pi), i)
-----------------------------------------(currentBeat + (i * 0.25)-----------
```

**Reply by isakube** [reply]
I don't think i0.25 works with lua

**Reply by glowsoony** [reply]
> I don't think i0.25 works with lua

maybe what should i do?

**Reply by glowsoony** [reply]
> I don't think i0.25 works with lua

DUDE it worked ! 
![image](https://user-images.githubusercontent.com/84847356/124504908-5ec90f80-dd96-11eb-9ae4-0106ecd63e63.png)


by changeing it to this function start (song)
    print("Succesfully initialized our song (Blammed)")
end

function Update (elapsed)
    local currentBeat = (songPos / 1000)(bpm/172)
    for i=0,7 do
        setActorY(defaultStrum0Y + 10 * math.cos(currentBeat + (i * 0.25) * math.pi), i)
    end
end

function beatHit (beat)
    print("Da beat has hit")
end

function septHit (beat)

end

function KeyPressed (key)

end

 the song worked but the wave of arrows affect dose not work anymore :(:(

**Reply by ACardboardBox1** [reply]
septHit.

**Reply by glowsoony** [reply]
> septHit.

IT WORKS!!! step hit a more was need and the code works FINE! 
![image](https://user-images.githubusercontent.com/84847356/124552681-dd0ecb80-de01-11eb-9193-455b953d7ecb.png)


new code! 
![image](https://user-images.githubusercontent.com/84847356/124553061-47277080-de02-11eb-8418-ea7f975e8fdc.png)
